hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -98,6 +98,15 @@ public boolean rename(Path src, Path dst) throws IOException {
         return renamed;
     }
 
+    @Override
+    public boolean delete(Path path, boolean recursive) throws IOException {
+        if (!fs.delete(path, recursive)) {
+            return false;
+        }
+        keyStore.remove(path.toString());
```

Comment:
can you leave a comment here about not being able to remove from the keystore all the time because we're not able to differentiate between file not found and the path being a directory?